gui/session/objectives/autociv_statsOverlay.js

```diff
@@ -73,34 +82,34 @@ AutocivControls.StatsOverlay = class
             this.update()
     }
 
-    indexDefeated(state)
+    indexNonDefeated(state)
     {
-        let indexListDefeated = []
+        let listNonDefeated = []
         for (let i = 0; i < state.length; i++)
-            if (state[i].state === "defeated")
-                indexListDefeated.push(i)
-        return indexListDefeated
+            if (state[i].state !== "defeated")
+                listNonDefeated.push(i)
+        return listNonDefeated
```

Comment:
Use filter, better legibility (if no performance loss)

Reply:
These are the once I tested. Sorted by slowest first, fastest last. The `for` loop is pretty good. :) 
```js
// 1 - reduce
return state.reduce((arr, value, index) => (value.state !== "defeated" ? [...arr, index] : arr), []);

// 2 - filter & map
return state.filter(x => x.state !== "defeated").map(x => x.playerNumber - 1)

// 3 - map & filter
return state.map((x, i) => x.state !== "defeated" ? i : "").filter(Number.isFinite)

// 4 - for loop
let listNonDefeated = []
for (let i = 0; i < state.length; i++)
    if (state[i].state !== "defeated")
        listNonDefeated.push(i)
return listNonDefeated
```

Measured with
```js
        const values = {}
        let startTime = Engine.GetMicroseconds()
        let listNonDefeated = this.indexNonDefeated(playerStates)
        warn(Engine.GetMicroseconds() - startTime)
        for (let stat of Object.keys(this.stats))
        {
            let list = playerStates.map(this.stats[stat])
            values[stat] = {
                "list": list,
                "min": this.minIndex(list, listNonDefeated),
                "max": this.maxIndex(list, listNonDefeated),
            }
        }

```